website/project/model.py

```diff
@@ -3563,3 +3572,159 @@ def _on_reject(self, user, token):
+    # DraftRegistration gets created. Files could be copied to a DraftRegistration
+    # rather than a Node, which would be much much cleaner in the event of a
+    # failure during archival.
+    # Additionally, future registration schemas will require users to select files
+    # that fulfill a certain requirement. Right now we will have to restrict those
+    # choices to OsfStorage files, but this is a non-issue if the third-party files
+    # have already been archived.
+    # storage = fields.ForeignField('osfstoragenodesettings')
+
+    # Dictionary field mapping
+    # { 'requiresApproval': true, 'fulfills': [{ 'name': 'Prereg Prize', 'info': <infourl> }]}
+    # config = fields.DictionaryField()
+
+    # Dictionary field mapping a draft's states during the review process to their value
+    # { 'isApproved': false, 'isPendingReview': false, 'paymentSent': false }
+    # flags = fields.DictionaryField()
+
+    def __init__(self, *args, **kwargs):
+        super(DraftRegistration, self).__init__(*args, **kwargs)
+        # TODO (samchrisinger): uncomment when we have flags/config
+        #meta_schema = self.registration_schema  # or kwargs.get('registration_schema')
+        #if meta_schema:
+        #    schema = meta_schema.schema
+        #    config = schema.get('config', {})
+        #    self.config = config
+        # TODO: uncomment to set flags
+        #     if not self.registration_schema:
+        #         flags = schema.get('flags', {})
+        #         for flag, value in flags.iteritems():
+        #             self.flags[flag] = value
+
+    # TODO: uncomment to expose approval/review properties
+    #    @property
+    #    def is_pending_review(self):
+    #        return self.flags.get('isPendingReview')
+    #
+    #    @is_pending_review.setter
+    #    def is_pending_review(self, value):
+    #        self.flags['isPendingReview'] = value
+    #
+    #    @property
+    #    def is_approved(self):
+    #        self.flags.get('isApproved')
+    #
+    #    @is_approved.setter
+    #    def is_approved(self, value):
+    #        self.flags['isApproved'] = value
+
+    def update_metadata(self, metadata, save=True):
+        # TODO: uncommnet to disallow editing drafts that are approved or pending approval
+        # if self.is_pending_review or self.is_approved:
+        #    raise NodeStateError('Cannot edit while this draft is being reviewed')
+        changes = []
+        for key, value in metadata.iteritems():
+            old_value = self.registration_metadata.get(key)
+            if not old_value or old_value.get('value') != value.get('value'):
+                changes.append(key)
+        self.registration_metadata.update(metadata)
+        if save:
+            self.save()
+        # TODO: uncomment to nullify approval state if edited
+        # project_signals.draft_edited.send(self, changes)
+        # self.after_edit(changes)
+        return changes
```

Comment:
In the future this will be used to see the User modifies the schema. If a user modifies a schema that has been approved (prereg specific feature), that approval is revoked. Since comments are embedded in  the data, we need a way to distinguish comments vs. data changes,